File: frontend/src/api/operatorLapangan.ts
import { apiClient } from "./client";

export interface ArmadaInfo {
  id: number;
  nama_armada?: string;
  no_polisi?: string;
  jenis_kendaraan_id?: number;
}

export interface OperatorLapangan {
  id: number;
  tenant_id: string;
  user_id: number;
  nip_nik: string;
  foto_url?: string;
  sim_file_url?: string;
  sim_expiry_date?: string;
  nama: string;
  role: string;
  armada?: ArmadaInfo;
  created_at: string;
  updated_at: string;
}

export interface OperatorLapanganCreate {
  nip_nik: string;
  nama: string;
  role: string;
  password?: string;
  foto_url?: string;
  sim_file_url?: string;
  sim_expiry_date?: string;
  armada_id?: number;
}

export interface OperatorLapanganUpdate {
  nama?: string;
  role?: string;
  password?: string;
  foto_url?: string;
  sim_file_url?: string;
  sim_expiry_date?: string;
  armada_id?: number;
}

export const getOperators = async () => {
  const { data } = await apiClient.get<OperatorLapangan[]>("/operator-lapangan/");
  return data;
};

export const getOperator = async (id: number) => {
  const { data } = await apiClient.get<OperatorLapangan>(`/operator-lapangan/${id}`);
  return data;
};

export const createOperator = async (payload: OperatorLapanganCreate) => {
  const { data } = await apiClient.post<OperatorLapangan>("/operator-lapangan/", payload);
  return data;
};

export const updateOperator = async (id: number, payload: OperatorLapanganUpdate) => {
  const { data } = await apiClient.put<OperatorLapangan>(`/operator-lapangan/${id}`, payload);
  return data;
};

File: backend/app/schemas/operator_lapangan.py
from datetime import date, datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict
from uuid import UUID


class OperatorLapanganBase(BaseModel):
    nip_nik: str
    foto_url: Optional[str] = None
    sim_file_url: Optional[str] = None
    sim_expiry_date: Optional[date] = None


class OperatorLapanganCreate(OperatorLapanganBase):
    nama: str
    role: str
    password: Optional[str] = None
    armada_id: Optional[int] = None


class OperatorLapanganUpdate(BaseModel):
    nama: Optional[str] = None
    role: Optional[str] = None
    password: Optional[str] = None
    foto_url: Optional[str] = None
    sim_file_url: Optional[str] = None
    sim_expiry_date: Optional[date] = None
    armada_id: Optional[int] = None


class ArmadaInfo(BaseModel):
    id: int
    nama_armada: Optional[str] = None
    no_polisi: Optional[str] = None
    jenis_kendaraan_id: Optional[int] = None

    model_config = ConfigDict(from_attributes=True)


class OperatorLapanganResponse(OperatorLapanganBase):
    id: int
    tenant_id: UUID
    user_id: int
    nama: str
    role: str
    armada: Optional[ArmadaInfo] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
